### PR TITLE
state: interface: Use new order book cache in interface methods

### PR DESCRIPTION
**Pull request by joeykraut**
### Purpose
This PR adds back in write paths for the order book cache and uses the new order book cache in the state interface. This involves two caller locations:
- The admin API, which roughly just uses renamed methods
- The matching engine, which now constructs an order book filter based on the order it is trying to match

### Testing
- [x] Unit tests pass
- [x] Testing in testnet with various match flows

### CODE REVIEW

File: state/src/caching/order_metadata_index.rs [hunk 1]
```diff
@@ -114,6 +118,10 @@ impl OrderMetadataIndex {
             sorted_vec.remove(idx);
         }
 
+        // Remove from the reverse index
+        let mut reverse_index = self.reverse_index.write().await;
+        reverse_index.remove(order_id);
+
```
Comment (akirillo) [inline state/src/caching/order_metadata_index.rs:124]: did we want to move this back under the fallible code?

File: state/src/replication/state_machine/snapshot.rs [hunk 2]
```diff
@@ -256,7 +256,8 @@ impl StateMachine {
         let order_cache_clone = self.applicator.config.order_cache.clone();
         let jh = tokio::task::spawn_blocking(move || {
             Self::copy_db_data(&snapshot_db, &db_clone)?;
-            tokio::spawn(async move {
+            let rt = tokio::runtime::Handle::current();
+            rt.block_on(async move {
```
Comment (akirillo) [inline state/src/replication/state_machine/snapshot.rs:260]: why do we need this to be blocking?

**Review by akirillo — APPROVED** [review]

File: state/src/caching/order_metadata_index.rs (hunk 1 above)
Comment (joeykraut) [reply state/src/caching/order_metadata_index.rs:124]: yes, the code above it depends on the entry still being in place

File: state/src/replication/state_machine/snapshot.rs (hunk 2 above)
Comment (joeykraut) [reply state/src/replication/state_machine/snapshot.rs:260]: because we're in a synchronous context

**Pull request merged**